parlai/tasks/wizard_of_internet/mutators.py

```diff
@@ -0,0 +1,255 @@
+                if found:
+                    out_episodes.append([e])
+            else:
+                pass
+        return out_episodes
+
+
+def chunk_docs_in_message(message, chunk_sz):
+    if CONST.RETRIEVED_DOCS not in message:
+        return message
+    new_message = message.copy()
+    docs = message.get(CONST.RETRIEVED_DOCS)
+    titles = message.get(CONST.RETRIEVED_DOCS_TITLES)
+    urls = message.get(CONST.RETRIEVED_DOCS_URLS)
+    new_docs = []
+    new_titles = []
+    new_urls = []
+    checked_sentences = list(
+        message.get(
+            CONST.SELECTED_SENTENCES,
+            message.get('labels', [CONST.NO_SELECTED_SENTENCES_TOKEN]),
+        )
+    )
+    for i in range(len(checked_sentences)):
+        checked_sentences[i] = checked_sentences[i].lstrip(' ').rstrip(' ')
+    if ' '.join(checked_sentences) == CONST.NO_SELECTED_SENTENCES_TOKEN:
+        checked_sentences = []
+    for ind in range(len(docs)):
+        d = docs[ind]
+        # Guarantees that checked sentences are not split in half (as we split by space).
+        for i in range(len(checked_sentences)):
+            d = d.replace(checked_sentences[i], "||CHECKED_SENTENCE_" + str(i) + "||")
+        while True:
+            end_chunk = d.find(' ', chunk_sz)
+            if end_chunk == -1:
+                # last chunk
+                for i in range(len(checked_sentences)):
+                    d = d.replace(
+                        "||CHECKED_SENTENCE_" + str(i) + "||", checked_sentences[i]
+                    )
+                new_docs.append(d)
+                new_titles.append(titles[ind])
+                new_urls.append(urls[ind])
+                break
+            else:
+                new_d = d[0:end_chunk]
+                for i in range(len(checked_sentences)):
+                    new_d = new_d.replace(
+                        "||CHECKED_SENTENCE_" + str(i) + "||", checked_sentences[i]
+                    )
+                new_docs.append(new_d)
+                new_titles.append(titles[ind])
+                new_urls.append(urls[ind])
+                d = d[end_chunk + 1 : -1]
+    new_message.force_set(CONST.RETRIEVED_DOCS, new_docs)
+    new_message.force_set(CONST.RETRIEVED_DOCS_TITLES, new_titles)
+    new_message.force_set(CONST.RETRIEVED_DOCS_URLS, new_urls)
+    return new_message
+
+
+@register_mutator("woi_chunk_retrieved_docs")
+class WoiChunkRetrievedDocs(MessageMutator):
+    """
+    Chunks '__retrieved-docs__' into smaller docs (max 100 words each).
+    """
+
+    @classmethod
+    def add_cmdline_args(
+        cls, parser: ParlaiParser, partial_opt: Optional[Opt] = None
+    ) -> ParlaiParser:
+        parser.add_argument(
+            '--woi-doc-chunk-size',
+            default=500,
+            type=int,
+            help='Document chunk size (in characters).',
+        )
+
+    def message_mutation(self, message: Message) -> Message:
+        chunk_sz = self.opt.get('woi_doc_chunk_size', 500)
```

Comment:
Given the mistakes that we had with setting flags before, may be we should have this as something like 

```.py
if not message.get('woi_doc_chunk_size'):
    logging.warning("The --woi-doc-chunk-size flag is received as None, using the default value of X")
```

It has happened to me that I set a flag without knowing that it was ignored. This at least warns about issue like that.

Reply:
well, what we actually need to do is fix the bug in parlai .. i was hoping kurt would do this...